### PR TITLE
LDAP auth

### PR DESCRIPTION
**Pull request by lGreenLightl**
### LDAP auth

Other implementation of LDAP auth with new features:

- _2 binding modes_
- _Multiple servers support_
- _TLS support (one-way and two-way)_

Also changed LDAP server docker image (currently supported) for feature tests

### CODE REVIEW

**Review by Denchick — APPROVED** [review]

File: pkg/auth/auth.go [hunk 1]
```diff
@@ -321,79 +318,59 @@ func AuthFrontend(cl client.Client, rule *config.FrontendRule) error {
 		err = cl.Send(&pgproto3.AuthenticationSASLFinal{Data: []byte(finalMsg)})
 		return err
 	case config.AuthLDAP:
-		if rule.AuthRule.LDAPConfig == nil {
```
Comment (EinKrebs) [inline pkg/auth/auth.go:324]: Why did you remove this check?

**Review by EinKrebs — CHANGES_REQUESTED** [review]

File: pkg/auth/auth.go (hunk 1 above)
Comment (lGreenLightl) [reply pkg/auth/auth.go:324]: I forgot to add it back when I removed previous code. Fixed

**Review by EinKrebs — APPROVED** [review]
